### PR TITLE
Gml 1690 chat history

### PR DESCRIPTION
**Pull request by RobRossmiller-TG**

### CODE REVIEW

File: chat-history/db/db.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+var (
+	db *gorm.DB
+	mu = sync.RWMutex{}
+)
+
+func createLogger(logPath string) logger.Interface {
```
Comment (parkererickson-tg) [inline chat-history/db/db.go:23]: Eventually this should follow the same logging format as the core CoPilot

File: chat-history/main.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		w.Write([]byte(`{"status":"OK"}`))
+	})
+
+	router.HandleFunc("GET /user/{userId}", routes.GetUserConversations)
+	router.HandleFunc("GET /conversation/{conversationId}", routes.GetConversation)
+	router.HandleFunc("POST /conversation", routes.UpdateConversation)
+
+	// create server with middleware
+	dev := strings.ToLower(os.Getenv("DEV")) == "true"
+	var port string
```
Comment (parkererickson-tg) [inline chat-history/main.go:35]: Do we want to hard-code ports? FWIW, this is what FastAPI is deployed on right now too...

Comment (RobRossmiller-TG) [reply chat-history/main.go:35]: Moved to config

Comment (RobRossmiller-TG) [reply chat-history/main.go:35]: They'll be in separate pods, so same port won't matter.

**Review by ahmed-tg — APPROVED** [review]
LGTM!  Nice job Rob!

**Review by parkererickson-tg — APPROVED** [review]
LGTM!

**Review by billshitg — APPROVED** [review]
I am gonna pretend I can read go - this looks good! (it actually looks great:)

**Pull request merged**